### PR TITLE
feat(patch): Add shared patch to expose activities of an app

### PR DESCRIPTION
**Pull request by bennett-sh**
This adds the patch as explained in ReVanced/revanced-patches-template#1259

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/all/activity/exportAll/patch/ExportAllActivitiesPatch.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                    val exportedAttribute = it.attributes.getNamedItem(EXPORTED_FLAG)
+                    if (exportedAttribute != null) {
+                        if(exportedAttribute.nodeValue != "true")
+                            exportedAttribute.nodeValue = "true"
+                        return@let
+                    }
+
+                    document.createAttribute(EXPORTED_FLAG)
+                        .apply { value = "true" }
+                        .let(it.attributes::setNamedItem)
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/all/activity/exportAll/patch/ExportAllActivitiesPatch.kt:33]: Why is this required in addition to the code above?

**Review by oSumAtrIX — APPROVED** [review]

File: src/main/kotlin/app/revanced/patches/all/activity/exportAll/patch/ExportAllActivitiesPatch.kt (hunk 1 above)
Comment (bennett-sh) [reply src/main/kotlin/app/revanced/patches/all/activity/exportAll/patch/ExportAllActivitiesPatch.kt:33]: Apps targeting a SDK below Android 12 are not required to include the attribute iirc so this code is for adding it if it doesn't already exist.

**Review by oSumAtrIX — APPROVED** [review]

**Comment by bennett-sh** [comment]
apps can mark activities as not exported. this means that only the system &
the app itself can launch them. this patch removes these restrictions
allowing apps like activity launcher to open them
